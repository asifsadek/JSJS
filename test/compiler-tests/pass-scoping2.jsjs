val x = "top level";

val f = /\(y) => {
    print("function_arg " ^ y);
    val x = "inner decl";
    print(x);
    (/\(x) => print("In anonymous function " ^ x);)("lambda");
    print("should be inner decl " ^ x);
};
f("funarg");
print("Back in top level scope " ^ x);
