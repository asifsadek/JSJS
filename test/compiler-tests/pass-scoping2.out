function_arg funarg
inner decl
In anonymous function lambda
should be inner decl inner decl
Back in top level scope top level
